src/nvim/api/private/helpers.c

```diff
@@ -808,6 +811,17 @@ static void init_type_metadata(Dictionary *metadata)
 
   PUT(*metadata, "types", DICTIONARY_OBJ(types));
 }
+static void init_api_level_metadata(Dictionary *metadata)
+{
+  Dictionary version = ARRAY_DICT_INIT;
+  PUT(version, "current", INTEGER_OBJ(NVIM_API_CURRENT));
+  PUT(version, "compatibility", INTEGER_OBJ(NVIM_API_COMPATIBILITY));
+#ifdef NVIM_VERSION_PRERELEASE
+  PUT(version, "prerelease", BOOLEAN_OBJ(true));
```

Comment:
If someone is developing a plugin against unreleased API, then they need to be following Neovim development closely to stay on top of any changes we may make before the API is released.  Therefore, they're aware that the API hasn't been released and should use it accordingly.

If they are then releasing the plugin that's using the unreleased API, then they need to pick up the cookie pieces if/when it crumbles.

I don't think providing extra metadata to help delineate whether the API is released or yet is necessary.  It's a technical measure for a social problem.